src/components/ebay-accordion/component-browser.ts

```diff
@@ -0,0 +1,38 @@
+import type { AttrBoolean, AttrString } from "marko/tags-html";
+import type { WithNormalizedProps } from "../../global";
+import type { Input as DetailsInput } from "../ebay-details/component-browser";
+
+export interface AccordionInput extends Omit<Marko.Input<"ul">, `on${string}`> {
+    size?: "regular" | "large";
+    "auto-collapse"?: AttrBoolean;
+    "a11y-role-description"?: AttrString;
+    items?: Marko.AttrTag<
+        Omit<DetailsInput, "size" | "alignment" | "layout" | `on${string}`>
+    >;
+    "on-toggle"?: (event: { originalEvent: Event; open: boolean }) => void;
+}
+
+export interface Input extends WithNormalizedProps<AccordionInput> {}
+class Accordion extends Marko.Component<Input> {
+    handleToggle(event: { originalEvent: Event; open: boolean }) {
+        
+        const { autoCollapse } = this.input;
+        if(autoCollapse && event.open) {
+            const accordion = this.getEl("accordion-root") as HTMLUListElement;
+            const detailsElements = accordion.querySelectorAll("details");
+            detailsElements.forEach((details: HTMLDetailsElement) => {
+                // Close the <details> element by setting its open attribute to false
+                if (details !== event.originalEvent.target) {
+                    details.open = false;
+                } 
+            });            
+        }
+        
+        this.emit("toggle", {
```

Comment:
Should the toggle event be triggered on the accordion? We could not have it and let the application adds on each detail. This will trigger twice for the auto collapse case (in case one is already open). Should we pass the index of the details that is being toggled to give more context?

```
onToggle({ open, index, originalEvent })
```

Reply:
Good catch, removed emitting the toggle event on accordion. I mean index works fine too, I am just being consistent with details for now.